### DISCUSSION TITLE
[Question] How customize default generated template?

### DISCUSSION THREAD

**Discussion by ghost** [gobuffalo/buffalo#2131] [Q&A]
For my projects I would like to make some changes in the default template generetaed by the cli command `buffano g r mymodel`.

I edited the file located in [here](https://github.com/gobuffalo/buffalo/tree/master/genny/resource/templates/core/templates/folder-name) but if a build the buffalo tool with `make install` and use it to generate resource it always use the default ones. 

How I can edit this templates?
Thanks

**Comment by paganotoni** [comment]
Hi @stormenergy91 there is no option currently to edit the model generation template.

**Comment by sio4** [comment]
@stormenergy91 , I think your question, your intention is about modifying the default templates and reinstall your own version of `buffalo` cli for internal use instead specifying specific templates on-the-fly with command line options or configuration. right? You already find where is the templates under the tree and tried to rebuild the binary after edit them.

In this case, you need to understand buffalo is embedding all text templates on the binary, and this is done by `packr` (now version 2, `packr2`) [1]. If you edit something, you need to run `packr2` in the root directory of the package then build and install the binary. You can find some autogenerated `.go` files were changed after you run the command, and they are a kind of zip storage for the templates.

FYI, currently `buffalo` cli has its own package [2] now, so I would suggest you to use that.

[1] https://github.com/gobuffalo/packr
[2] https://github.com/gobuffalo/cli
